runtime/runtime/src/actions.rs

```diff
@@ -83,13 +83,10 @@ pub(crate) fn get_code_with_cache(
     state_update: &TrieUpdate,
     account_id: &AccountId,
     account: &Account,
-) -> Result<Arc<ContractCode>, String> {
+) -> Result<Option<Arc<ContractCode>>, StorageError> {
     debug!(target:"runtime", "Calling the contract at account {}", account_id);
     let code_hash = account.code_hash;
-    let code = || {
-        get_code(state_update, account_id)
-            .ok_or_else(|| format!("cannot find contract code for account {}", account_id.clone()))
-    };
+    let code = || get_code(state_update, account_id);
     crate::cache::get_code_with_cache(code_hash, code)
```

Comment:
This is the only place where we use `crate::cache::get_code_with_cache` should it take `code_hash, state_update, account_id` arguments instead of passing it a closure?

Reply:
this way the cache logic has no dependencies